modules/runtime/viewer/pygame_viewer.py

```diff
@@ -32,36 +39,55 @@ def __init__(self, params=None, **kwargs):
             self.screen = None
             print("No available video device")
 
+    def drawMap(self, map):
```

Comment:
drawMap is already defined in the base viewer (modules/runtime/viewer/viewer.py) and thus should not be defined here. What was your motivation here?

Reply:
I saw the map was being drawn again and again, which is not necessary and is one of the two main reasons for the performance drop.
To avoid redrawing, we need to have a "surface" drawn with the map, and "blit" back to the main screen during each running frame.

https://github.com/bark-simulator/bark/blob/7bc54de159edf12368e0994be1a481d743fccfa5/modules/runtime/viewer/pygame_viewer.py#L68
https://github.com/bark-simulator/bark/blob/7bc54de159edf12368e0994be1a481d743fccfa5/modules/runtime/viewer/pygame_viewer.py#L72
I think avoiding redrawing cannot be done without these two lines, and cannot be done with drawline2d, that's why I defined a new drawmap

Reply:
can't we use the base class method? es for example in https://stackoverflow.com/questions/805066/call-a-parent-classs-method-from-child-class

Reply:
I think the logic of drawMap in pygame_viewer could be too complicated to adapt to the function in the base class... as to not redraw the map, we have to draw to a different source other than the screen, and calculate the coordinates to display in the screen
Or should I remove the feature of not redrawing the map?